command/agent/consul/int_test.go

```diff
@@ -31,7 +31,6 @@ func testLogger() *log.Logger {
 // TestConsul_Integration asserts TaskRunner properly registers and deregisters
 // services and checks with Consul using an embedded Consul agent.
 func TestConsul_Integration(t *testing.T) {
-	t.Skip("-short set; skipping")
```

Comment:
Tracked it down. I was calling `testtask.Path()` without setting the environment which caused these tests to be re-executed and killed... when the erroneously re-executed tests were killed it would leak a Consul.

(This was as fun to debug as it sounds.)